### PR TITLE
[WIP] `Search by keyword` filter dismiss other filters

### LINKED ISSUE #1301: `Search by keyword` filter dismiss other filters
### Preconditions (*)
Magento 2.4-develop
Adobe-Stock-Integration 1.1-develop

1. Installed Magento with Adobe Stock Integration
2. Configured integration in Stores -> Configuration -> Advanced-> System -> Adobe Stock Integration fieldset

### Steps to reproduce (*)

1. Upload image with name `a4`
2. Search this image by search input
image
3. Apply filter `Source` - `Adobe Stock`
image

### Expected result (*)

Image can't be found because it was uploaded

### Actual result (*)

Image is found

### PR DESCRIPTION
**Pull request by engcom-Charlie**
### Description (*)

Closes #1301

### Fixed Issues (if relevant)

1. magento/adobe-stock-integration#1301: `Search by keyword` filter dismiss other filters
2. ...

### Manual testing scenarios (*)

1. ...
2. ...

### CODE REVIEW

File: MediaGalleryUi/Model/Listing/DataProvider.php [hunk 1]
```diff
@@ -84,4 +100,29 @@ public function getData(): array
...
+
+    /**
+     * Add fulltext filter to SearchCriteriaBuilder and reset SearchCriteria
+     *
+     * @return void
+     */
+    private function prepareSearchCriteria(): void
+    {
+        if ($this->fulltextFilter) {
+            $this->searchCriteria = null;
```
Comment (sivaschenko) [inline MediaGalleryUi/Model/Listing/DataProvider.php:124]: What is the reason of clearing search criteria and custom flow for the fulltext filter?

**Review by sivaschenko — CHANGES_REQUESTED** [review]
Thanks for the pull request @engcom-Charlie ! Please see my question

File: MediaGalleryUi/Model/Listing/DataProvider.php (hunk 1 above)
Comment (engcom-Charlie) [reply MediaGalleryUi/Model/Listing/DataProvider.php:124]: Thank You for comment, in selectModifier we get condition for Fulltext

Comment (engcom-Charlie) [reply MediaGalleryUi/Model/Listing/DataProvider.php:124]: so it breaks Select

**Pull request closed without merging**